build/fpga-regmap/src/lib.rs

```diff
@@ -186,7 +231,13 @@ fn write_reg_fields(children: &[Node], prefix: &str, output: &mut String) {
     }
 }
 
-fn write_node(node: &Node, prefix: &str, output: &mut String) {
+fn write_node(
+    parents: Vec<String>,
+    node: &Node,
+    prefix: &str,
+    output: &mut String,
+) {
+    let mut new_parents = parents.clone();
```

Comment:
AFAICT, this function never references `parents` again after cloning it. Since we take `parents` by value, we can just push the new parents into `parents`, because we have exclusive access to it --- I think the clone isn't necessary here, unless I've missed something.

Reply:
alternatively, since only some cases in `write_node` actually need to push to parents, the most efficient thing would be to change `write_node` to take `parents: &Vec<String>`,  remove the clone in `recurse_reg_map` and pass it by reference, and then only clone `parents` in the cases where we need to push a new parent to pass to a function (so that we avoid the unconditional clone in the `Node::Mem` case).

again, optimizing for performance here is not terribly important, but...that's what i'd do, so i figured i'd suggest it.

Reply:
That makes sense to me. I shall make it so!